src/runtime_src/core/tools/xbflash2/OO_Program_Qspips.cpp

```
@@ -0,0 +1,295 @@
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream>
+#include <fstream>
+
+namespace {
+
+int 
+qspips_flash(po::variables_map& vm) {
+    std::string bdf;
+    std::string flash_type;
+    std::string bin_file;
+    std::vector<std::string> bin_files;
+    std::string soffset;
+    size_t offset = 0;
+    bool force = false;
+    std::string sBar, sBarOffset;
+    size_t baroff = INVALID_OFFSET;
+    int bar = 0;   
+
+    XBU::sudo_or_throw_err();
+     
+    //mandatory command line args
+    try {
+        bdf = vm["device"].as<std::string>();
+        bin_files = vm["image"].as<std::vector<std::string>>();
+    }
+    catch (...) {
+        throw std::errc::invalid_argument;
+    }
+
+    //optional command line args
+    try {
+        if (vm.count("flash-part"))
+            flash_type = vm["flash-part"].as<std::string>();
+        if (vm.count("offset")) {
+            soffset = vm["offset"].as<std::string>();
+            if (!soffset.empty()) {
+                std::stringstream sstream(soffset);
+                sstream >> offset;
+            }
+        }
+        if (vm.count("force"))
+            force = true;
+    }
+    catch (...) {
+    }
+
+    //optionals command line args
+    try {
+        if (vm.count("bar")) {
+            sBar = vm["bar"].as<std::string>();
+            if (!sBar.empty())
+                bar = std::stoi(sBar);
+        }
+        if (vm.count("bar-offset")) {
+            sBarOffset = vm["bar-offset"].as<std::string>();
+            if (!sBarOffset.empty()) {
+                std::stringstream sstream(sBarOffset);
+                sstream >> baroff;
+            }
+        }
+    }
+    catch (...) {
+    }
+
+    if (bdf.empty()) {
+        std::cout << "Error: Please provide mgmt BDF\n";
+        throw std::errc::invalid_argument;
+    }
+
+    if (!bin_files.size()) {
+        std::cout << "Error: Please provide proper BIN file.\n";
```

If this is an error, please use std::err. Also consider throwing the exception instead of printing it.